### PR TITLE
completed assignment 1 and labs 1-3

### PR DESCRIPTION
**Pull request by ian-lilley**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
adding code to complete assignment 1 and labs 1-3

## What did you learn from the changes you have made?
I learned how to make a convolutional neural network

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
none

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
there were many challenges. the greatest of which is that we were not taught how to do any of this. i overcame it by using chatgpt and websites to teach myself. 
the second greatest challenge was the amount of time required to complete the assignment and three labs. to overcome this, i didn't sleep all weekend

## How were these changes tested?
the code was run and compared to expected outcomes

## Checklist
- [ x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by x-rojas-io — APPROVED** [review]
Hi Ian!

Thank you for submitting your DL assignments.

Your A1 and Lab 1-3 are now COMPLETE.

Lab-1 comments
Missing Questions

Lab-2 comments
Missing c) Look at worst prediction errors

A1 comments:
1b- Correctly normalizes the image data and converts labels to one-hot encoding. However, you are splitting the training set into x_train_new and x_validate, but the names for the training and validation data (x_train_new, y_train_new, x_validate, y_validate) are inconsistent with the previous naming of x_train, y_train. It’s better to either directly rename them or to use consistent variable names for clarity.
2b- AUC is more suitable for binary classification, for multiclass classification you can use ‘categorical_accuracy’


Thanks,
Nestor[LS]
